consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -530,17 +530,24 @@ public boolean isDescendantOf(final BlockHeader ancestorBlock, final BlockHeader
         newBlock.getNumber(),
         newBlock.getBlockHash());
 
-    if (ancestorBlock.getBlockHash().equals(newBlock.getHash())) {
+    Optional<BlockHeader> parentOf =
```

Comment:
because this is a public method that gets called from fcU, we need to check whether ancestorblock == newBlock, OR we can just start with newBlock instead of its parent.. i.e. parentBlock = Optional.of(newBlock), and then loop